mlir/lib/Target/LLVMIR/Dialect/GENX/GENXToLLVMIRTranslation.cpp

```diff
@@ -65,38 +65,64 @@ class GENXDialectLLVMIRTranslationInterface
   LogicalResult
   amendOperation(Operation *op, NamedAttribute attribute,
                  LLVM::ModuleTranslation &moduleTranslation) const final {
-    if (attribute.getName() == GENX::GENXDialect::getKernelFuncAttrName()) {
-      auto func = dyn_cast<LLVM::LLVMFuncOp>(op);
-      if (!func)
+    auto func = dyn_cast<LLVM::LLVMFuncOp>(op);
+    if (!func)
+      return failure();
+
+    llvm::LLVMContext &llvmContext = moduleTranslation.getLLVMContext();
+    llvm::Function *llvmFunc = moduleTranslation.lookupFunction(func.getName());
+
+    // Set max_work_group_size metadata.
+    if (attribute.getName() ==
+        GENX::GENXDialect::getMaxWorkGroupSizeAttrName()) {
+      auto value = attribute.getValue().dyn_cast<ArrayAttr>();
+      if (!value)
         return failure();
 
-      // For GPU kernels, set SPIR_KERNEL calling convention.
-      llvm::Function *llvmFunc =
-          moduleTranslation.lookupFunction(func.getName());
-      llvmFunc->setCallingConv(llvm::CallingConv::SPIR_KERNEL);
```

Comment:
We do not need to set the calling convention here. This code should handle the metadata only.

Reply:
setCallingConv is needed for Intel and AMD. Intel GPU uses spir_kernel and AMD uses AMD_GPU_KERNEL.   NVVM uses LLVM "kernel" MDNode under "nvvm.annotations" MetaData to mark kernel functions.

Reply:
Yes I understand that a kernel function needs to have the `spir_kernel` qualifer. We set that qualifier in PR  https://github.com/intel-sandbox/triton-dse/pull/55 before we attach these attributes to the function. This function IMO should just deal with lowering the dialect attributes. Now, it is true that these attribute make sense only for a kernel function, so we could add an assertion here to make sure the calling convention of the function is `spir_kernel`.

Reply:
GENX dialect will support multiple frontends.  It is better for Triton to set the GENX::GENXDialect::getKernelFuncAttrName() and let the GENX lowering to set the calling convention.  Otherwise, every frontend targeting GENX will need to set the calling convention.

Reply:
ok, will do.